src/drivers/uart/src/amdmmio.rs

```diff
@@ -0,0 +1,136 @@
+/*
+ * This file is part of the oreboot project.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 2 of the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+use core::ops;
+use model::*;
+
+use register::mmio::{ReadOnly, ReadWrite};
+use register::register_bitfields;
+
+const RETRY_COUNT: u32 = 100_000;
```

Comment:
This is a pretty common retry count for uarts, which is where I got it. It is useful because it is so large that it makes it detectable that you are always timing out when you watch uart output. Were you to make it much closer to a reasonable value you might miss that you're not testing the right bits in the register, or that they are not operating correctly. 

Not making this up: it's caught this mistake before for me and it caught it recently in our i8250 driver where we were in fact always timing out and not checking the right bits. The performance is so bad it stands out. You don't want timeout performance to be good.